### ISSUE TITLE
[datasets/documentation] Add 1-2 dataset examples + visualized targets 

### ISSUE THREAD

**Issue by felixdittrich92** [mindee/doctr#729]
### 🚀 The feature
Adding the possibility to inspect a loaded dataset with a dataset viewer

### Motivation, pitch
```
from doctr.datasets import IC03
train_set = IC03(train=True, download=True)
train_set.inspect()
-> open a view (matplotlib !?) to inspect the loaded dataset with corresponding boxes and labels (for both: if exist / very similar to result.show(doc))
```

Why ?
It would be a nice feature to give the user a look into the loaded dataset (for some reasons (trust, interest if dataset is not known, ..))
Makes it easier to check new implemented datasets (also for contributer)

Additional

- show all examples from set as thumbnail grid where image to open can be choose
- or only single examples (iteratable 1 - n)
- ? @fg-mindee

**Comment by fg-mindee** [comment]
Hi @felixdittrich92 :wave: 

Would you mind specifying what you have in mind please (editing the issue description) ? "clickable data set view" ?

In the motivation, pitch section, please provide some code sample and the type of expected result you would like to see :pray:

**Comment by felixdittrich92** [comment]
@fg-mindee 
yes sorry for the lazy description :sweat_smile: 
Hopefully now it is more understandable !? :dancers:

**Comment by felixdittrich92** [comment]
@fg-mindee 
Any feedback ? Do you agree to implement this feature ?

**Comment by fg-mindee** [comment]
Hey @felixdittrich92 :wave: 

My apologies! Considering the scope of the project, I don't think this should be integrated into the library as-is:
- this will lead to other discussions of other visualization-related features, which are dependencies/topics quite different from the ones used for OCR inference
- for training purposes, we already have a `--show-samples` flag that displays a batch of samples
- for documentation, we could add illustrations of image + target
- but for the dataset code, I'd argue it should stick to its core need: providing a fast data reading tool for training/evaluation purposes

More generally, if people are interested in the dataset, they can freely go explore the references we have added in the doc. Now that we have harmonized DS, on top of this, the `--show-samples` features is already providing enough I think :+1:

**Comment by felixdittrich92** [comment]
@fg-mindee 
Yes i agree it was only a idea to give the user a better overview :) 
But to improve the documentation with 1-2 examples + visualized targets is a really good way to do this also great idea 👍

**Comment by felixdittrich92** [comment]
closed by #857
